TACC/Core-CMS-Custom
# task/WP-345: Entities changes UTH & task/WP-376: Add 'Year' column to registration admin listing

## Overview
Addresses requested changes to entities on the registration form + adds 'Year' column to admin registration listing
…

## Related

- [WP-345](https://jira.tacc.utexas.edu/browse/WP-345)
- [WP-376](https://jira.tacc.utexas.edu/browse/WP-376)
…

## Changes
**WP-345**
- For file type checkboxes, now at least one of three (Medical, Pharmacy, Dental) must be selected to continue to submit form
- Entity coverage estimate fields are now text fields which only accept ints or float (where needed) by regex patterns
- Slightly increased state box to accommodate larger state abbreviations + added JS listeners to swap between state abbr. and full name, depending on user interaction on drop down

**WP-376**
- Added 'Year' column to admin list registrations table
- Consolidated styling for registrations listings to one CSS file + changed renewal listing to grab styles from there instead
- Added `registration_year` to database query for creating new registrations
   - * For renewals, this year is indexed up by 1 to account for renewals for the next registration period
…

## Testing

1. Go to https://localhost:8000/register/request-to-submit
2. Test the form as normal, and confirm that:
    - On the State drop down, the default value should now be blank
    - On selecting a choice in the drop down, only the corresponding state abbreviation shows up in the box
    - Click away from the drop down and then click back and confirm that the option you selected displays both the 
       abbreviation and full state name, as before selection
    - On the entities, on any number of entity entries, that you cannot submit the form unless one of Medical, Pharmacy, and 
       Dental boxes is selected
    - On the Coverages Estimates section, that any input other than ints on the first two fields and two-decimal-point floats 
      (dollars + cents) on the final field raises an error (str input is allowed but not accepted on submit)
    - After successful submission, go to https://localhost:8000/administration/list-registration-requests and confirm the 
       record shows on the listing with the current year on the table, and all the info on View/Edit record is accurate
    - On the Edit Record action, confirm the same tests as above on the registration form
3. Go to https://localhost:8000/register/list-registration-requests and first confirm table looks the same as before, then initiate a registration renewal
4. On the renewal, fill in the necessary info and submit. On successful submission, navigate back to the administration registration listings and confirm the renewal record shows up with the year as 2024

## UI

…
